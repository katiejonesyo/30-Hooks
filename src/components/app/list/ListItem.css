.image {
    height: 20%;
    width: 20%;
    border-radius: 10px;
    box-shadow: black 10px 10px 30px 1px;
  }