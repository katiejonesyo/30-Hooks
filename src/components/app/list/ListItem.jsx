import React from 'react';
import PropTypes from 'prop-types';
import styles from './ListItem.css';

function ListItem({ name, photoUrl }) {
  return (
    <>
      <h1>{name}</h1>
      <img className={styles.image} src={photoUrl} alt="avatar character" />
    </>
  );
}

ListItem.propTypes = {
  name: PropTypes.string,
  photoUrl: PropTypes.string,
};

export default ListItem;


