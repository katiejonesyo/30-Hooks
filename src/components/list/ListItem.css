
  h1 {
    font-size: 2.5rem;
    text-decoration: 'none';
    padding-bottom: 5%;
    font-family: fantasy;
  }

  main {
      margin-top: 0%;
      background-color: rgb(65, 47, 33);
      color: rgb(20, 11, 6);
      padding: 3rem;
      text-decoration: 'none';
      padding-bottom: 10px;

  }
  
  ul {
    text-decoration: 'none';
  }

 .image {
    margin: 0rem;
    box-shadow: rgb(218, 173, 144) 0px 0px 30px 30px;
  }

  .link {
    text-decoration: 'none';
  }


  
  

