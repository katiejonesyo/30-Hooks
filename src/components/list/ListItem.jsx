import React from 'react';
import PropTypes from 'prop-types';
import styles from './ListItem.css';

function ListItem({ name, photoUrl }) {
  return (
    <>
    <main>
      <h1>{name}</h1>
      <img className={styles.image} src={photoUrl} alt="avatar character" />
      </main>
    </>
  );
}

ListItem.propTypes = {
  name: PropTypes.string,
  photoUrl: PropTypes.string,
};

export default ListItem;


