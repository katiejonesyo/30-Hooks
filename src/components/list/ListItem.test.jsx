import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ListItem from '../list/ListItem';

describe('ListItem component', () => {
  afterEach(() => cleanup());
  it('renders a single character', () => {
    const { asFragment } = render(
      <ListItem name="some name" photoUrl="some photo url" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});



