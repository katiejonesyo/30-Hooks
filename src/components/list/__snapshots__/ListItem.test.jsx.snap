// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders a single character 1`] = `
<DocumentFragment>
  <main>
    <h1>
      some name
    </h1>
    <img
      alt="avatar character"
      class="image"
      src="some photo url"
    />
  </main>
</DocumentFragment>
`;
